spec/models/spree/payment/processing_decorator_spec.rb

```diff
@@ -14,5 +14,12 @@
 
       expect(payment.close!).to be_truthy
     end
+
+    it 'payment receives close! method' do
+      payment = create(:amazon_payment, state: 'completed')
+
+      expect(payment).to receive(:close!)
```

Comment:
When I do this I get:

`1) Spree::Payment::Processing#close! delegates close! to the payment source
     Failure/Error: client.close_order_reference(@amazon_order_reference_id)

     WebMock::NetConnectNotAllowedError:
       Real HTTP connections are disabled. Unregistered request: POST https://mws.amazonservices.com/OffAmazonPayments_Sandbox/2013-01-01 with body 'AWSAccessKeyId=&Action=CloseOrderReference&AmazonOrderReferenceId=ORDER_REFERENCE&SellerId=&SignatureMethod=HmacSHA256&SignatureVersion=2&Timestamp=2016-09-27T18%3A56%3A12Z&Version=2013-01-01&Signature=rTTOuhl2xyp4zmDEwCRacq58I08V26kbRTR%2FxBMucyA%3D' with headers {'Accept'=>'*/*', 'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3', 'User-Agent'=>'Language=Ruby; ApplicationLibraryVersion=1.1.0; Platform=x86_64-darwin14; MWSClientVersion=2013-01-01; ApplicationName=; ApplicationVersion='}

       You can stub this request with the following snippet:

       stub_request(:post, "https://mws.amazonservices.com/OffAmazonPayments_Sandbox/2013-01-01").
         with(:body => "AWSAccessKeyId=&Action=CloseOrderReference&AmazonOrderReferenceId=ORDER_REFERENCE&SellerId=&SignatureMethod=HmacSHA256&SignatureVersion=2&Timestamp=2016-09-27T18%3A56%3A12Z&Version=2013-01-01&Signature=rTTOuhl2xyp4zmDEwCRacq58I08V26kbRTR%2FxBMucyA%3D",
              :headers => {'Accept'=>'*/*', 'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3', 'User-Agent'=>'Language=Ruby; ApplicationLibraryVersion=1.1.0; Platform=x86_64-darwin14; MWSClientVersion=2013-01-01; ApplicationName=; ApplicationVersion='}).
         to_return(:status => 200, :body => "", :headers => {})

       ============================================================
     # /Users/allisonreilly/.rvm/gems/ruby-2.2.1/gems/webmock-2.1.0/lib/webmock/http_lib_adapters/net_http.rb:114:in `request'
     # /Users/allisonreilly/.rvm/gems/ruby-2.2.1/gems/pay_with_amazon-1.1.0/lib/pay_with_amazon/request.rb:82:in `post'
     # /Users/allisonreilly/.rvm/gems/ruby-2.2.1/gems/pay_with_amazon-1.1.0/lib/pay_with_amazon/request.rb:48:in `send_post'
     # /Users/allisonreilly/.rvm/gems/ruby-2.2.1/gems/pay_with_amazon-1.1.0/lib/pay_with_amazon/client.rb:870:in `operation'
     # /Users/allisonreilly/.rvm/gems/ruby-2.2.1/gems/pay_with_amazon-1.1.0/lib/pay_with_amazon/client.rb:735:in `close_order_reference'
     # ./lib/amazon_mws.rb:142:in `close_order_reference'
     # ./app/models/spree_amazon/order.rb:25:in `close_order_reference!'
     # ./app/models/spree/amazon_transaction.rb:71:in `close!'
     # ./app/models/spree/payment/processing_decorator.rb:6:in `close!'
     # ./spec/models/spree/payment/processing_decorator_spec.rb:32:in `block (3 levels) in <top (required)>'`